x-pack/plugins/security_solution/public/exceptions/manage_exceptions/create_shared_exception_list.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useState, useCallback, useRef, useEffect } from 'react';
+import {
+  EuiFlyout,
+  EuiTitle,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiText,
+  EuiFieldText,
+  EuiSpacer,
+  EuiTextArea,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlexItem,
+} from '@elastic/eui';
+import type { HttpSetup } from '@kbn/core-http-browser';
+import type { ErrorToastOptions, Toast, ToastInput } from '@kbn/core-notifications-browser';
+import { i18n as translate } from '@kbn/i18n';
+
+import { useCreateSharedExceptionListWithOptionalSignal } from './use_create_shared_list';
+import {
+  CREATE_SHARED_LIST_TITLE,
+  CREATE_SHARED_LIST_NAME_FIELD,
+  CREATE_SHARED_LIST_DESCRIPTION,
+  CREATE_BUTTON,
+  CLOSE_FLYOUT,
+  CREATE_SHARED_LIST_DESCRIPTION_PLACEHOLDER,
+  CREATE_SHARED_LIST_NAME_FIELD_PLACEHOLDER,
+} from './translations';
+
+export const CreateSharedListFlyout = memo(
+  ({
+    handleRefresh,
+    http,
+    handleCloseFlyout,
+    addSuccess,
+    addError,
+  }: {
+    handleRefresh: () => void;
+    http: HttpSetup;
+    addSuccess: (toastOrTitle: ToastInput, options?: unknown) => Toast;
+    addError: (error: unknown, options: ErrorToastOptions) => Toast;
+    handleCloseFlyout: () => void;
+  }) => {
+    const [listName, setListName] = useState('');
+    const [description, setDescription] = useState('');
+
+    const { start: createSharedExceptionList, ...createSharedExceptionListState } =
+      useCreateSharedExceptionListWithOptionalSignal();
+    const ctrl = useRef(new AbortController());
+
+    const onListNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      setListName(e.target.value);
+    };
+    const onDescriptionChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setDescription(e.target.value);
+    };
```

Comment:
One of the best practices is to define a form object with the fields that shape your domain obj and use one `onChange` as below 

```
 const [newListDetails, setNewListDetails] = useState<ListDetails>(listDetails);
 const onChange = ({ target }: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = target;
    setNewListDetails({ ...newListDetails, [name]: value });
  };
```

You can actually use the ListDetails from `@kbn/securitysolution-exception-list-components`

Reply:
Fixed here: f5f3ec39fea6438b0485f85e75afa38a8da209dc